content/docs/tasks/traffic-management/egress-tls-origination.md

```diff
@@ -0,0 +1,195 @@
+
+    ```bash
+        cat <<EOF | istioctl create -f -
+        apiVersion: networking.istio.io/v1alpha3
+        kind: ServiceEntry
+        metadata:
+          name: cnn
+        spec:
+          hosts:
+          - "*.cnn.com"
+          ports:
+          - number: 80
+            name: http-port
+            protocol: HTTP
+          - number: 443
+            name: https-port
+            protocol: HTTPS
+        EOF
+    ```
+
+1.  Make a request to the external HTTP service:
+
+    ```command
+    $ kubectl exec -it $SOURCE_POD -c sleep -- curl -IL http://edition.cnn.com/politics
+    HTTP/1.1 301 Moved Permanently
+    ...
+    location: https://edition.cnn.com/politics
+    ...
+
+    HTTP/1.1 200 OK
+    Content-Type: text/html; charset=utf-8
+    ...
+    Content-Length: 151654
+    ...
+    ```
+
+    The output should be similar to the above (unimportant details replaced by ellipsis):
+
+Note the `-L` flag of `curl`. It instructs `curl` to follow redirects. In this case,
+the server responded with a redirect response ([301 Moved Permanently](https://tools.ietf.org/html/rfc2616#section-10.3.2)) to an HTTP request to http://edition.cnn.com/politics. The redirect response instructs the client to send an additional request, this time by HTTPS to https://edition.cnn.com/politics. For the second request, the server responds with the requested content and _200 OK_ status code.
+
+While for the user of `curl` this redirection happens transparently, there are two issues here. The first issue is the redundant first request, which doubles the latency of fetching the content of http://edition.cnn.com/politics. The second issue is that the path of the URL, _politics_ in this case, is sent in clear text. If there is an attacker who sniffs the communication between our application and _cnn.com_, the attacker would know which specific topics and articles of _cnn.com_ our application fetched. Due to the privacy reasons we may want to prevent such disclosure from the attacker.
+
+In the next section we configure Istio to perform TLS origination to resolve the two issues above. Let's clean our configuration before proceeding to the next section:
+
+```command
+$ istioctl delete serviceentry cnn
+```
+
+## TLS origination for Egress traffic
+
+1.  Define a `ServiceEntry` to allow traffic to _edition.cnn.com_, a `VirtualService` to perform request port rewriting, and a `DestinationRule` for TLS origination. This time use HTTP protocol for the port 443 of the `ServiceEntry`, since the clients will use HTTP protocol and Istio will perform TLS origination for them. Note two additional changes in our `VirtualService`. First, we have to use a specific host `edition.cnn.com`, this is because the Envoy proxy has to know exactly which host and port it should access by HTTPS. The second change is `resolution` `DNS`, the change is required according to Envoy's configuration specifics.
```

Comment:
This paragraph is still confusing. How about this:

1. Define a ServiceEntry to allow traffic to edition.cnn.com, a VirtualService to perform request port rewriting,
   and a DestinationRule for TLS origination. 

   Unlike the ServiceEntry in the previous section, here we use HTTP for the protocol on port 433, since clients
   will  send HTTP requests and Istio will perform TLS origination for them. Also, the resolution must be set
   to DNS to correctly configure Envoy in this case.

   Finally, note that the VirtualService uses a specific host edition.cnn.com (no wildcard) because the Envoy
   proxy needs to know exactly which host and port to access using HTTPS.

Reply:
@frankbu OK, let me handle this tomorrow.